ztools/x509/x509.go

```diff
@@ -778,6 +778,10 @@ func (c *Certificate) CheckSignatureFrom(parent *Certificate) (err error) {
 
 	// TODO(agl): don't ignore the path length constraint.
 
+	if parent.Subject.String() != c.Issuer.String() {
+		return errors.New("Mis-match issuer/subject")
+	}
+
```

Comment:
Actually I take this back. While this function is named *terribly*, there _are_ other checks in it already. :(